server/src/main/java/org/elasticsearch/operator/OperatorHandler.java

```diff
@@ -0,0 +1,118 @@
+ * the basic contract for implementing cluster state update handlers in operator mode.
+ */
+public interface OperatorHandler<T> {
+    String CONTENT = "content";
+
+    /**
+     * The operator handler name is a unique identifier that is matched to a section in a
+     * cluster state update content. The operator cluster state updates are done as a single
+     * cluster state update and the cluster state is typically supplied as a combined content,
+     * unlike the REST handlers. This name must match a desired content key name in the combined
+     * cluster state update, e.g. "ilm" or "cluster_settings" (for persistent cluster settings update).
+     *
+     * @return a String with the operator key name, e.g "ilm".
+     */
+    String name();
+
+    /**
+     * The transform method of the operator handler should apply the necessary changes to
+     * the cluster state as it normally would in a REST handler. One difference is that the
+     * transform method in an operator handler must perform all CRUD operations of the cluster
+     * state in one go. For that reason, we supply a wrapper class to the cluster state called
+     * {@link TransformState}, which contains the current cluster state as well as any previous keys
+     * set by this handler on prior invocation.
+     *
+     * @param source The parsed information specific to this handler from the combined cluster state content
+     * @param prevState The previous cluster state and keys set by this handler (if any)
+     * @return The modified state and the current keys set by this handler
+     * @throws Exception
+     */
+    TransformState transform(Object source, TransformState prevState) throws Exception;
+
+    /**
+     * Sometimes certain parts of the cluster state cannot be created/updated without previously
+     * setting other cluster state components, e.g. composable templates. Since the cluster state handlers
+     * are processed in random order by the OperatorClusterStateController, this method gives an opportunity
+     * to any operator handler to declare other operator handlers it depends on. Given dependencies exist,
+     * the OperatorClusterStateController will order those handlers such that the handlers that are dependent
+     * on are processed first.
+     *
+     * @return a collection of operator handler names
+     */
+    default Collection<String> dependencies() {
+        return Collections.emptyList();
+    }
+
+    /**
+     * All implementations of OperatorHandler should call the request validate method, by calling this default
+     * implementation. To aid in any special validation logic that may need to be implemented by the operator handler
+     * we provide this convenience method.
+     *
+     * @param request the master node request that we base this operator handler on
+     */
+    default void validate(MasterNodeRequest<?> request) {
```

Comment:
Does this differ across any operator handler implementations? If not, could it just be a local utility method in the operator code?